src/types/nitro.ts

```diff
@@ -113,6 +113,20 @@ export interface NitroOptions extends PresetOptions {
   _config: NitroConfig
 
   // General
+  preset:
+    | 'aws-lambda'
+    | 'azure'
+    | 'cleavr'
+    | 'cloudflare'
+    | 'digital-ocean'
+    | 'firebase'
+    | 'heroku'
+    | 'layer0'
+    | 'netlify'
+    | 'render-com'
+    | 'stormkit'
+    | 'vercel'
+    | (string & {})
```

Comment:
can we infer from the actual list of presets - e.g. `keyof typeof presets`?

Reply:
Sure! How would you approach the difference between camel case and dash case though?

![image](https://user-images.githubusercontent.com/48835293/197292665-9eb0dfe7-4f2e-4d45-817a-facc4d2c2c3a.png)

Reply:
How about excluding presets with capitals like this?

```ts
type DeCamel<T extends string> = T extends Lowercase<T> ? T : never
```

We could also write a slightly more complex type to _convert_ camel -> kebab, but I think that's probably overkill.

Reply:
That works great, but does it mean we would only autocomplete preset names without capitals?

Reply:
Threw this together for fun. And definitely not to use here, but ...

```ts
type UppercaseLetter = 'A' | 'B' | 'C' | 'D' | 'E' | 'F' | 'G' | 'H' | 'I' | 'J' | 'K' | 'L' | 'M' | 'N' | 'O' | 'P' | 'Q' | 'R' | 'S' | 'T' | 'U' | 'V' | 'W' | 'X' | 'Y' | 'Z'
type KebabCase<T extends string> = T extends Lowercase<T>
  ? T
  : T extends `${infer Start}${UppercaseLetter}${string}`
    ? Start extends Lowercase<Start>
      ? T extends `${Start}${infer U}${infer Rest}`
        ? Start extends ''
          ? `${Lowercase<U>}${KebabCase<Rest>}`
          : `${Start}-${Lowercase<U>}${KebabCase<Rest>}`
        : never
      : never
    : never
```

Reply:
Oh, good point... might be worth just hard-coding 😆

Reply:
That was quick! :)

Found a shorter one which also seems to work fine, any thoughts?
https://stackoverflow.com/questions/66140451/typescript-add-kebab-case-types-form-actual-camel-case-keys
```
type Kebab<T extends string, A extends string = ""> =
    T extends `${infer F}${infer R}` ?
    Kebab<R, `${A}${F extends Lowercase<F> ? "" : "-"}${Lowercase<F>}`> :
    A
```
![image](https://user-images.githubusercontent.com/48835293/197295491-bf8cd375-3552-40b6-a2de-c0f6d8055348.png)

Reply:
Oh, nice! Seems a much more reasonable level of complexity.